lib/at_shell/at_shell.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <shell/shell.h>
+#include <nrf_modem_at.h>
+#include <modem/at_monitor.h>
+
+static struct shell *global_shell;
+static const char at_usage_str[] =
+	"Usage: at <subcommand>\n"
+	"\n"
+	"Subcommands:\n"
+	"  events_enable     Enable AT event handler which prints AT notifications\n"
+	"  events_disable    Disable AT event handler\n"
+	"\n"
+	"Any other subcommand is interpreted as AT command and sent to the modem.\n";
+
+AT_MONITOR(at_shell_monitor, ANY, at_notif_handler, PAUSED);
+
+static void at_notif_handler(const char *response)
+{
+	shell_print(global_shell, "AT event handler:\n%s", response);
+}
+
+static void at_print_error_info(const struct shell *shell, int error)
+{
+	switch (nrf_modem_at_err_type(error)) {
+	case NRF_MODEM_AT_ERROR:
+		shell_print(shell, "ERROR");
+		break;
+	case NRF_MODEM_AT_CME_ERROR:
+		shell_print(shell, "CME ERROR: %d", nrf_modem_at_err(error));
+		break;
+	case NRF_MODEM_AT_CMS_ERROR:
+		shell_print(shell, "CMS ERROR: %d", nrf_modem_at_err(error));
+		break;
+	default:
+		break;
```

Comment:
The default branch should never get executed, I guess, but this could be combined with NRF_MODEM_AT_ERROR, so it would at least print an error.
```suggestion
	switch (nrf_modem_at_err_type(error)) {
	case NRF_MODEM_AT_CME_ERROR:
		shell_print(shell, "CME ERROR: %d", nrf_modem_at_err(error));
		break;
	case NRF_MODEM_AT_CMS_ERROR:
		shell_print(shell, "CMS ERROR: %d", nrf_modem_at_err(error));
		break;
	case NRF_MODEM_AT_ERROR:
	default:
		shell_print(shell, "ERROR");
		break;
```

Reply:
Good point, added as suggested